rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/MtuWatcher.java

```diff
@@ -0,0 +1,45 @@
+package com.polidea.rxandroidble.internal.connection;
+
+
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.inject.Inject;
+import javax.inject.Named;
+import rx.Completable;
+import rx.CompletableSubscriber;
+import rx.Subscription;
+import rx.functions.Action1;
+
+@ConnectionScope
+class MtuWatcher extends Completable implements MtuProvider {
+
+    private final AtomicInteger currentMtuAtomicInteger;
```

Comment:
Does it have to be atomic? Do we expect concurrent updates from different threads?

Reply:
It has to be a reference so it could be referenced from the `OnSubscribe`